caddyconfig/httpcaddyfile/httptype.go

```diff
@@ -376,45 +376,65 @@ func (st *ServerType) serversFromPairings(
 				return nil, fmt.Errorf("server block %v: compiling matcher sets: %v", sblock.block.Keys, err)
 			}
 
-			hosts := sblock.hostsFromKeys(false, false)
+			var httpHosts, httpsHosts []string
+			for _, key := range sblock.block.Keys {
+				addr, err := ParseAddress(key)
```

Comment:
Keys are already parsed in the latest master; iterate sblock.keys instead.

Reply:
Nice thanks!